pkg/relayer/session/persistence.go

```diff
@@ -0,0 +1,200 @@
+package session
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+
+	"github.com/pokt-network/poktroll/pkg/relayer"
+	prooftypes "github.com/pokt-network/poktroll/x/proof/types"
+	sharedtypes "github.com/pokt-network/poktroll/x/shared/types"
+)
+
+// populateSessionTreeMap reconstructs the in-memory session tree map from previously persisted sessions.
+// This method:
+// 1. Retrieves all persisted session metadata from storage
+// 2. Validates each session against current blockchain height to determine its lifecycle state
+// 3. Deletes expired sessions that are past their proof submission window
+// 4. Deletes sessions past their claim window that don't have an associated onchain claim
```

Comment:
supernit: I feel like this would read slightly better if the ordering of the clauses of one was reversed to match the other:

```suggestion
// 3. Deletes expired sessions that are past their proof submission window
// 4. Deletes sessions that don't have an associated onchain claim which are past their claim window 
```

Alternatively:
```suggestion
// 3. Deletes expired sessions that are past their proof submission window
// 4. Deletes unclaimed sessions that are past their claim creation window
```
OR
```suggestion
// 3. Deletes expired sessions whose proof submission window has elapsed
// 4. Deletes unclaimed sessions whose claim creation window has elapsed
```